src/asciidoctor-kroki.js

```diff
@@ -97,6 +101,46 @@ const processKroki = (processor, parent, attrs, diagramType, diagramText, contex
   }
 }
 
+/**
+ * @param {string} diagramText
+ * @param { {vfs: { read: function} } } context
+ * @returns {string}
+ */
+function preprocessVegaLite (diagramText, context) {
```

Comment:
We should probably move this function into a dedicated file and export it. It would allow to write unit tests.

Reply:
Good idea, I will have a look at it tomorrow.

Reply:
I moved the function to a new file and added some tests. There is a "TODO" in "test\preprocess.spec.js" concerning inlining referenced remote files. Could you please have a look at it?